packages/babel-plugin/src/utils/evaluate-expression.tsx

```diff
@@ -205,6 +206,90 @@ const traverseFunction = (expression: t.Function, meta: Metadata) => {
   return createResultPair(value as t.Expression, updatedMeta);
 };
 
+/**
+ * Will find the function node for the call expression and wrap an IIFE around it (to avoid name collision)
+ * and move all the parameters mapped to passed arguments in the IIFE's scope (own scope in this case).
+ * It will also set own scope path so that when we recursively evaluate any node,
+ * we will look for its binding in own scope first, then parent scope.
+ *
+ * @param expression Expression we want to interrogate.
+ * @param meta Meta data used during the transformation.
+ */
+const traverseCallExpression = (expression: t.CallExpression, meta: Metadata) => {
+  const callee = expression.callee;
+  let value: t.Node | undefined | null = undefined;
+  let updatedMeta: Metadata = meta;
+
+  if (t.isExpression(callee)) {
+    let functionNode = null;
+
+    if (t.isIdentifier(callee)) {
+      const resolvedBinding = resolveBindingNode(callee.name, updatedMeta);
+
+      if (resolvedBinding && resolvedBinding.constant) {
+        functionNode = resolvedBinding.node;
+      }
+    } else if (t.isMemberExpression(callee) && t.isIdentifier(callee.property)) {
+      const { accessPath, bindingIdentifier } = getMemberExpressionMeta(callee);
+
+      if (bindingIdentifier) {
+        const resolvedBinding = resolveBindingNode(bindingIdentifier.name, updatedMeta);
+
+        if (resolvedBinding && resolvedBinding.constant) {
+          if (t.isObjectExpression(resolvedBinding.node)) {
+            functionNode = getValueFromObjectExpression(resolvedBinding.node, accessPath);
+          }
+        }
+      }
+    }
+
+    if (functionNode && t.isFunction(functionNode)) {
+      const { params } = functionNode;
+
+      const evaluatedArguments = expression.arguments.map(
+        (argument) => evaluateExpression(argument as t.Expression, updatedMeta).value
+      );
+
+      updatedMeta.parentPath.traverse({
+        CallExpression(callExpressionPath) {
+          // Only look for expression for which we are evaluating
+          if (callExpressionPath.node === expression) {
```

Comment:
are we doing this just to get ahold of the expression path? could use use `getPathForExpression` instead?

Reply:
Ah! here were are just checking the reference. There can be many `CallExpression` under that `expression`, and we are only interested in nodes when `CallExpression` is same as passed `expression`